includes/Templates/Received.php

```
@@ -88,12 +93,19 @@ public function get_template(): void {
-			$status_setting   = $gateway_settings['gateways'][ $payment_method ]['order_status'] ?? 'wc-completed';
-			$completed_status = 'wc-' === substr( $status_setting, 0, 3 ) ? substr( $status_setting, 3 ) : $status_setting;
-			$order_complete   = 'pos-open' !== $completed_status;
+			$order_json = $this->get_order_json( $order->get_id() );
+			// Emit only when the order is genuinely beyond payment. needs_payment() covers
+			// every status WooCommerce or a gateway still considers payable — pending, failed,
+			// and (via our own woocommerce_valid_order_statuses_for_payment filter) pos-open /
+			// pos-partial — so an async gateway that redirects before the provider confirms
+			// emits nothing. The explicit POS-status exclusion stays because needs_payment()
+			// is false for a ZERO-TOTAL order regardless of status: a parked zero-total cart
+			// must not report itself as a completed payment.
+			$order_complete = ! $order->needs_payment() && ! \in_array( $order->get_status(), array( 'pos-open', 'pos-partial' ), true );
```

PHPMD (2.15.0)

[warning] Avoid unused local variables such as '$order_complete'. (undefined)

(UnusedLocalVariable)